### PR TITLE
GKE improvements for D4science

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Several fixes to the spawner and authenticator to better work on GKE with D4Science.
Namely:
- Avoid error 500 when context is not available
- Add namespaces as option to spawner
- Add image override option to spawner
- Remove tornado setting from the spawner
- Add label as authentication option

---

**Related issue :**

### CODE REVIEW

File: egi_notebooks_hub/d4science.py [hunk 1]
```diff
@@ -63,6 +72,13 @@ class D4ScienceOauthenticator(GenericOAuthenticator):
         help="""The URL for getting DataMiner resources from the
                 Information System of D4science""",
     )
+    d4science_label_name = Unicode(
+        "d4science-namespace",
```
Comment (andrea-manzi) [inline egi_notebooks_hub/d4science.py:76]: this should bevd4science-label?

Comment (andrea-manzi) [reply egi_notebooks_hub/d4science.py:76]: or better d4science-label-name?

Comment (enolfc) [reply egi_notebooks_hub/d4science.py:76]: you mean the default value?

Comment (andrea-manzi) [reply egi_notebooks_hub/d4science.py:76]: yes, is that correct?

Comment (enolfc) [reply egi_notebooks_hub/d4science.py:76]: yes, this is a key-value pair, the key name is defined by this `d4science_label_name` which defaults to `d4science-namespace` and then the value is whatever comes from the authenticator

**Review by andrea-manzi — APPROVED** [review]
LGTM!

**Pull request merged**